src/Agent.Worker/Build/GitCommandManager.cs

```diff
@@ -181,8 +215,12 @@ public async Task LoadGitExecutionInfo(IExecutionContext context, bool useBuiltI
                 // The Windows agent ships a copy of Git
                 if (PlatformUtil.RunningOnWindows)
                 {
-                    _gitPath = Path.Combine(HostContext.GetDirectory(WellKnownDirectory.Externals), "git", "cmd", $"git{IOUtil.ExeExtension}");
-                    _gitLfsPath = Path.Combine(HostContext.GetDirectory(WellKnownDirectory.Externals), "git", PlatformUtil.BuiltOnX86 ? "mingw32" : "mingw64", "bin", "git-lfs.exe");
+                    context.Debug("Git paths are resolving from internal dependencies");
+
+                    var localGitPaths = GetInternalGitPaths(context, AgentKnobs.UseLatestGitVersion.GetValue(context).AsBoolean());
+
+                    _gitPath = localGitPaths.Item1;
```

Comment:
Small suggestion - Item1 and Item2 can be confused with each other, so maybe we can use ValueTuple and refer to returned values by their names? (here and in a few other places)

Reply:
Good idea.
Changed.